src/test/java/mate/academy/onlinebookstore/controller/BookControllerTest.java

```diff
@@ -0,0 +1,197 @@
+                .apply(springSecurity())
+                .build();
+        teardown(dataSource);
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/book/create-table-books_categories.sql")
+            );
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/category/add-two-categories.sql")
+            );
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/book/add-two-books.sql")
+            );
+        }
+    }
+
+    @AfterAll
+    static void afterAll(@Autowired DataSource dataSource) {
+        teardown(dataSource);
+    }
+
+    @SneakyThrows
+    static void teardown(DataSource dataSource) {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/book/remove-all-books.sql")
+            );
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/category/remove-all-categories.sql")
+            );
+        }
+    }
+
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    @Test
+    @DisplayName("Save new book.")
+    void createBook_ValidRequestDto_Ok() throws Exception {
+        CreateBookRequestDto requestDto = new CreateBookRequestDto();
+        requestDto.setTitle("Effective Java");
+        requestDto.setAuthor("Joshua Bloch");
+        requestDto.setIsbn("978-0134685991");
+        requestDto.setPrice(BigDecimal.valueOf(49.99));
+        requestDto.setDescription("A comprehensive guide to best practices in Java programming.");
+        requestDto.setCategoryIds(List.of(1L, 2L));
+
+        BookDto expected = new BookDto();
+        expected.setTitle(requestDto.getTitle());
+        expected.setAuthor(requestDto.getAuthor());
+        expected.setIsbn(requestDto.getIsbn());
+        expected.setPrice(requestDto.getPrice());
+        expected.setDescription(requestDto.getDescription());
+
+        String json = objectMapper.writeValueAsString(requestDto);
+
+        MvcResult result = mockMvc.perform(
+                        post("/books")
+                                .content(json)
+                                .contentType(MediaType.APPLICATION_JSON)
+                )
+                .andExpect(status().isCreated())
+                .andReturn();
+
+        BookDto actual = objectMapper.readValue(
+                result.getResponse().getContentAsString(),
+                BookDto.class);
+
+        Assertions.assertNotNull(actual);
+        EqualsBuilder.reflectionEquals(expected, actual);
```

Comment:
Better use static imports. Please pay attention to the fact that reflectionEquals returns a boolean value. It doesn’t  work as you expected without `assetTrue()` (fix it everywhere)